src/lib/withFolders.jsx

```diff
@@ -0,0 +1,50 @@
+import { useState, useEffect } from 'react'
+import { Q } from 'cozy-client'
+
+import logger from 'lib/logger'
+
+const FILES_DOCTYPE = 'io.cozy.files'
+export const ROOT_FOLDER_ID = 'io.cozy.files.root-dir'
+const TRASH_DIR_ID = 'io.cozy.files.trash-dir'
+const buildFoldersQuery = ({ currentFolderId }) =>
+  Q(FILES_DOCTYPE)
+    .where({
+      dir_id: currentFolderId,
+      type: 'directory',
+      name: { $gt: null }
+    })
+    .partialIndex({
+      _id: {
+        $ne: TRASH_DIR_ID
+      }
+    })
+    .indexFields(['dir_id', 'type', 'name'])
+    .sortBy([{ dir_id: 'asc' }, { type: 'asc' }, { name: 'asc' }])
+
+const useFolders = client => {
+  const [loading, setLoading] = useState(false)
+  const [failed, setFailed] = useState(false)
+  const [folders, setFolders] = useState([])
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        setLoading(true)
+        const foldersQuery = buildFoldersQuery({
+          currentFolderId: ROOT_FOLDER_ID
+        })
+        const fetchedFolders = await client.queryAll(foldersQuery)
```

Comment:
We don't like the use of queryAll(). It can be very slow even if I don't think we'll have a lot of cases with thousands and thousands of folders. 

The way we handle that is to make the first `query`. Then,  the query returns an `hasMore` attribute. If this value if `true` then we should call `fetchMore()` only if needed. And when I say that, it means, we create an Intersection Observer https://developer.mozilla.org/fr/docs/Web/API/Intersection_Observer_API

We have a Load More button that do that for you: https://github.com/cozy/cozy-ui/blob/master/react/LoadMore/index.jsx

And you can see its usage in Drive here: https://github.com/cozy/cozy-drive/blob/master/src/drive/web/modules/views/Folder/FolderViewBody.jsx#L212-L214